ironic/tests/unit/drivers/modules/drac/test_inspect.py

```diff
@@ -378,3 +388,170 @@ def test__get_pxe_dev_nics_with_empty_list(self, mock_get_drac_client):
+
+        system_mock.processors.summary = '8', 'MIPS'
+
+        system_mock.simple_storage.disks_sizes_bytes = (
+            1 * units.Gi, units.Gi * 3, units.Gi * 5)
+        system_mock.storage.volumes_sizes_bytes = (
+            2 * units.Gi, units.Gi * 4, units.Gi * 6)
+
+        system_mock.ethernet_interfaces.summary = {
+            '00:11:22:33:44:55': sushy.STATE_ENABLED,
+            '66:77:88:99:AA:BB': sushy.STATE_DISABLED}
+
+        return system_mock
+
+    def test_get_properties(self):
+        expected = redfish_utils.COMMON_PROPERTIES
+        driver = drac_inspect.DracRedfishInspect()
+        self.assertEqual(expected, driver.get_properties())
+
+    @mock.patch.object(redfish_utils, 'get_system', autospec=True)
+    @mock.patch.object(DracRedfishInspect, '_get_mac_address', autospec=True)
+    def test__get_pxe_dev_nics_with_UEFI_boot_mode(
+            self, mock__get_mac_address, mock_get_system):
+        system_mock = self.init_system_mock(mock_get_system.return_value)
+        system_mock.boot.mode = 'uefi'
+        expected_pxe_nic = ['24:6E:96:70:49:00']
+        mock__get_mac_address.return_value = '24:6E:96:70:49:00'
+
+        with task_manager.acquire(self.context, self.node.uuid,
+                                  shared=True) as task:
+            task.driver.inspect.inspect_hardware(task)
+            pxe_dev_nics = task.driver.inspect._get_pxe_dev_nics(task)
+            self.assertEqual(expected_pxe_nic, pxe_dev_nics)
+
+    @mock.patch.object(requests, 'get', autospec=True)
+    @mock.patch.object(requests, 'post', autospec=True)
+    @mock.patch.object(redfish_utils, 'get_system', autospec=True)
+    @mock.patch.object(DracRedfishInspect, '_get_mac_address', autospec=True)
+    def test__get_pxe_dev_nics_with_BIOS_boot_mode(
+            self, mock__get_mac_address, mock_get_system, mock_post, mock_get):
+        system_mock = self.init_system_mock(mock_get_system.return_value)
+        system_mock.boot.mode = 'bios'
+        expected_pxe_nic = ['24:6E:96:70:49:00']
+        mock__get_mac_address.return_value = '24:6E:96:70:49:00'
+        mock_post.return_value.__dict__ = {'status_code': 202, 'headers': {
+            'Location': '/redfish/v1/TaskService/Tasks/JID_704694755409'}}
+        mock_get.return_value.__dict__ = {
+            '_content': '<SystemConfiguration Model="PowerEdge R640" '
+            'ServiceTag="7T1NHK2" TimeStamp="Wed Oct 16 11:38:17 2019"> \n '
+            '<Component FQDD="NIC.Slot.1-1-1"> \n '
+            '<Attribute Name="LegacyBootProto">NONE</Attribute> \n '
+            '</Component> \n <Component FQDD="NIC.Integrated.1-1-1"> \n '
+            '<Attribute Name="LegacyBootProto">PXE</Attribute> \n '
+            '</Component> \n </SystemConfiguration> '}
+
+        with task_manager.acquire(self.context, self.node.uuid,
+                                  shared=True) as task:
+            task.driver.inspect.inspect_hardware(task)
```

Comment:
I don't think the line above is needed since the line below is calling the _get_pxe_dev_nics() method directly.

Reply:
Done